df_py/util/graphutil.py

```diff
@@ -6,10 +7,47 @@
 def submit_query(query: str, chainID: int) -> dict:
     subgraph_url = networkutil.chain_id_to_subgraph_uri(chainID)
     request = requests.post(subgraph_url, "", json={"query": query}, timeout=30)
+
     if request.status_code != 200:
         # pylint: disable=broad-exception-raised
         raise Exception(f"Query failed. Return code is {request.status_code}\n{query}")
 
     result = request.json()
 
     return result
+
+
+def get_last_block(chain_id: int) -> int:
+    query = "{_meta { block { number } } }"
+    result = submit_query(query, chain_id)
+
+    return result["data"]["_meta"]["block"]["number"]
+
+
+def wait_to_block(chain_id: int, block_number: int):
```

Comment:
`wait_to_latest_block()` and `wait_to_block()` are very similar wrt final behavior. And the names don't reflect the difference. This is confusing.

And the only place that `wait_to_block()` is called is from `wait_to_latest_block()`.

Let's make this less confusing.

Options:
1. Move all code from `wait_to_block()` into `wait_to_latest_block()`, and deprecate `wait_to_block()`
2. Rename the methods where the name clarifies their differences. (And put a `_` as the first character of `wait_to_block()`.
3. Other?

I recommend (1) if it's possible. And I think it is.

----
Second comment: this function (or functions) needs docstrings and unit tests.